plugin-dev/Source/Sentry/Private/SentrySubsystem.cpp

```diff
@@ -154,11 +163,25 @@ void USentrySubsystem::Close()
 	if(GLog && OutputDevice)
 	{
 		GLog->RemoveOutputDevice(OutputDevice.Get());
+		OutputDevice = nullptr;
 	}
 
 	if(GError && OutputDeviceError)
 	{
+		if(OnAssertDelegate.IsValid())
+		{
+			OutputDeviceError->OnError.Remove(OnAssertDelegate);
+			OnAssertDelegate.Reset();
+		}
+
 		GError = OutputDeviceError->GetParentDevice();
+		OutputDeviceError = nullptr;
+	}
+
+	if(OnEnsureDelegate.IsValid())
+	{
+		FCoreDelegates::OnHandleSystemEnsure.Remove(OnEnsureDelegate);
+		OnEnsureDelegate.Reset();
```

Comment:
Should this go into the closure above as well?

Reply:
No, in order to capture ensures we don't use any of our custom output devices but rather rely on the engine's `FCoreDelegates::OnHandleSystemEnsure` delegate.